lib/lightning_web/controllers/changeset_json.ex

```diff
@@ -15,4 +16,124 @@ defmodule LightningWeb.ChangesetJSON do
   def error(%{changeset: %Ecto.Changeset{} = changeset}) do
     %{errors: error(changeset)}
   end
+
+  def provisioning_error(%{changeset: %Ecto.Changeset{} = changeset}) do
+    %{errors: traverse_provision_errors(changeset)}
+  end
+
+  # inspired by: https://github.com/elixir-ecto/ecto/blob/v3.11.2/lib/ecto/changeset.ex#L4037
+  defp traverse_provision_errors(%Ecto.Changeset{errors: errors} = changeset) do
+    errors
+    |> Enum.reverse()
+    |> merge_keyword_keys()
+    |> merge_related_keys(changeset)
+  end
```

Comment:
Why not recursively use `traverse_errors`? I mean I see you said it was inspired by, could we not use it in this case - and recurse or does it expect atom keys or something?

Reply:
`traverse_errors` is focused on the `error message` and not the `error key`. It keeps that `error key` to itself